accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,298 @@
+
+  - operate on the to-be master or worker node:
+    - Copy the SSH key of the Management Node for passwordless login in the node. In case of existing machines, the SSH file must be copied (e.g. via Salt), whereas the AutoYaST case will be already covered during provisioning.
+    - Assign the CaaS Platform channels assigned via Salt (SUSE CaaS Platform Extension to SUSE Linux Enterprise 15 and SUSE Containers Module 15 SP1 x86_64), whereas the AutoYaST case will be already covered during provisioning.
+    - De-register the machine from SUSE Manager (if it is registered)
+
+  - operate on the Management Node:
+    - Salt will invoke `skuba` (using `cmd.run`) to bootstrap the first master or make the selected node join the cluster. The user must specify which user to use for passwordless login (in this case `sles`) and whether to use `sudo` or not.
+
+    Example:
+
+    ```
+    bootstrap_first_master:
+      cmd.run:
+        - name: skuba node bootstrap --user sles --sudo --target <node IP> <node name>
+    
+    bootstrap_additional_master:
+      cmd.run:
+        - name: skuba node join --role master --user sles --sudo --target <node IP> <node name>
+    
+    bootstrap_worker:
+      cmd.run:
+        - name: skuba node join --role worker --user sles --sudo --target <node IP> <node name>
+    ```
+
+    The output of the `skuba` command must be parsed and shown to the user in SUSE Manager in case of errors.
+
+## Manage a CaaS Platform cluster using SUSE Manager
+
+### Show cluster status
+
+SUSE Manager shows the status of the cluster by invoking `skuba cluster status` on the Management Node (via Salt).
+
+E.g. of the output:
+
+```
+NAME         OS-IMAGE                              KERNEL-VERSION        CONTAINER-RUNTIME   HAS-UPDATES   HAS-DISRUPTIVE-UPDATES
+master0   SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+worker0   SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+```
+
+The output is presented in the "CaaS Platform > cluster name" section.
+
+### Removing nodes from the cluster
+
+A node can also be removed from the cluster:
+
+- temporarily (cordoned off): using `kubectl` from the Management Node via Salt.
+- permanently: by invoking `skuba node remove <nodename>` via Salt
+
+Example state:
+
+```
+remove_node:
+      cmd.run:
+        - name: skuba node remove <nodename>
+```
+
+NOTE: Other `skuba` options can be implemented in the future (e.g. resetting and reconfiguring a node).
+
+### Cluster upgrade
+
+All the cluster updates must be handled by `skuba`.
+
+SUSE Manager can check and notify the user if an update for the cluster is present by triggering `skuba cluster upgrade plan` on the Management Node (using Salt) and check if a new output is available in a cron fashion (e.g. Taskomatic job).
```

Comment:
Not strictly to be run from the management node, as mentioned previously.

Reply:
Fixed